### ISSUE TITLE
aws_mq_configuration revision increment not showing in plan

### ISSUE THREAD

**Issue by askainet** [terraform-providers/terraform-provider-aws#14842]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.12.29
+ provider.aws v3.3.0
```

### Affected Resource(s)

* aws_mq_configuration

### Terraform Configuration Files

```hcl
resource "aws_mq_configuration" "this" {
  name           = "testing"
  description    = "MQ broker configuration for testing"
  engine_type    = "ActiveMQ"
  engine_version = "5.15.12"
  data           = file("${path.module}/amazonmq_config.xml")
  tags = {
    Tag = timestamp()
  }
}
```

### Expected Behavior

Terraform does not create a new aws_mq_configuration revision when only updating tags, OR
if AWS is who actually creates a new aws_mq_configuration revision when updating tags, then the provider must reflect that and show the revision increment.

### Actual Behavior

A new aws_mq_configuration revision is created, despite the plan not showing it:

1. `terraform apply`

```
  # aws_mq_configuration.this will be updated in-place
  ~ resource "aws_mq_configuration" "this" {
        ...redacted...
        latest_revision = 1
      ~ tags            = {
          - "Tag" = "2020-08-25T20:07:59Z"
        } -> (known after apply)
    }
```

2. `terraform apply` again, and we find revision was indeed incremented despite not being shown in previous plan

```
  # aws_mq_configuration.this will be updated in-place
  ~ resource "aws_mq_configuration" "this" {
        ...redacted...
        latest_revision = 2
      ~ tags            = {
          - "Tag" = "2020-08-25T20:08:30Z"
        } -> (known after apply)
    }
```

This generates drift when referring this aws_mq_configuration in a aws_mq_broker resource, and even makes a `terraform apply -auto-approve` fail with error:

```
Error: Provider produced inconsistent final plan

When expanding the plan for aws_mq_broker.this to include new values learned
so far during apply, provider "registry.terraform.io/-/aws" produced an
invalid new value for .configuration.revision: was cty.NumberIntVal(1), but
now cty.NumberIntVal(2).
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Same behaviour occurs using provider version 2.70.0

**Comment by ewbankkit** [comment]
@askainet Thanks for raising this issue.
You are correct that updating just the tags on the resource causes the configuration to be updated also:

https://github.com/terraform-providers/terraform-provider-aws/blob/eae0bc7053091ab78f24c8a98da5011977d4c353/aws/resource_aws_mq_configuration.go#L158-L170

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.5.0 of the Terraform AWS Provider, later next week. Thanks to @ewbankkit for the implementation. 👍
